cowprotocol/accounting/rewards/mainnet/vouch_registry_query_1541516.sql

```diff
@@ -0,0 +1,104 @@
+with
+
+last_block_before_timestamp as (
+    select max(number) from ethereum.blocks
+    where time < cast('{{end_time}}' as timestamp)
+),
+
+-- Query Logic Begins here!
+vouches as (
+    select
+        evt_block_number,
+        evt_index,
+        solver,
+        cowRewardTarget as reward_target,
+        pool,
+        sender,
+        True as active
+    from cow_protocol_ethereum.VouchRegister_evt_Vouch
+    inner join query_4056263
+        on
+            pool = bondingPool
+            and sender = initial_funder
+    where evt_block_number <= (select * from last_block_before_timestamp)
+),
+
+invalidations as (
+    select
+        evt_block_number,
+        evt_index,
+        solver,
+        Null as reward_target,  -- This is just ot align with vouches to take a union
+        pool,
+        sender,
+        False as active
+    from cow_protocol_ethereum.VouchRegister_evt_InvalidateVouch
+    inner join query_4056263
+        on
+            pool = bondingPool
+            and sender = initial_funder
+    where evt_block_number <= (select * from last_block_before_timestamp)
+),
+
+-- At this point we have excluded all arbitrary vouches (i.e. those not from initial funders of recognized pools)
+-- This ranks (solver, pool, sender) by most recent (vouch or invalidation)
+-- and yields as rank 1, the current "active" status of the triplet.
+vouches_and_invalidations as (
+    select *
+    from vouches
+    union distinct
+    select * from invalidations
```

Comment:
styling looks inconsistent

Reply:
oh thought this would have been fixed automatically. Sooorry